Here's the error:

```
--- Building tests for swift ---
+ /usr/bin/cmake --build /home/asb/Developer/swift/build/Ninja-DebugAssert/swift-linux-x86_64 -- -j24 SwiftUnitTests
[4/9] Linking CXX executable unittests/Driver/SwiftDriverTests
FAILED: unittests/Driver/SwiftDriverTests
: && /usr/bin/clang++   -Wno-unknown-warning-option -Werror=unguarded-availability-new -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -std=c++11 -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wcovered-switch-default -Wdelete-non-virtual-dtor -Wstring-conversion -fcolor-diagnostics -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -g  -latomic -fuse-ld=lld unittests/Driver/CMakeFiles/SwiftDriverTests.dir/DependencyGraphTests.cpp.o  -o unittests/Driver/SwiftDriverTests -L/home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/./lib -Wl,-rpath,/home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/./lib /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSupport.a -lpthread /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libgtest_main.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libgtest.a -lpthread lib/libswiftDriver.a -lpthread lib/libswiftFrontend.a lib/libswiftMigrator.a lib/libswiftIDE.a lib/libswiftFrontend.a lib/libswiftMigrator.a lib/libswiftIDE.a lib/libswiftParseSIL.a lib/libswiftSIL.a lib/libswiftSema.a lib/libswiftSerialization.a lib/libswiftClangImporter.a lib/libswiftParse.a lib/libswiftIndex.a lib/libswiftAST.a lib/libswiftMarkup.a /home/asb/Developer/swift/build/Ninja-DebugAssert/cmark-linux-x86_64/src/libcmark.a lib/libswiftSyntax.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMDebugInfoDWARF.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86CodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86AsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86Desc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86AsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86Disassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86Info.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMX86Utils.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMAsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMDisassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMDesc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMAsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMInfo.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMARMUtils.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64CodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMGlobalISel.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64AsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64Disassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64Desc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64AsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64Info.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAArch64Utils.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCAsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCDesc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCAsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCDisassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPowerPCInfo.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZAsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZDisassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZDesc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZAsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSystemZInfo.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMDebugInfoCodeView.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMDebugInfoMSF.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSelectionDAG.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsAsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsDesc.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsAsmPrinter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsDisassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMCDisassembler.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMipsInfo.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangFrontendTool.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMCoroutines.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMCoverage.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMLTO.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMObjCARCOpts.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMPasses.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMipo.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMIRReader.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAsmParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMInstrumentation.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMLinker.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMCodeGen.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMBitWriter.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMScalarOpts.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMInstCombine.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMVectorize.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMTransformUtils.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMTarget.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMAnalysis.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMObject.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangARCMigrate.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangStaticAnalyzerFrontend.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangStaticAnalyzerCheckers.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangStaticAnalyzerCore.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangCrossTU.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangIndex.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangFormat.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangToolingCore.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangASTMatchers.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangRewriteFrontend.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangFrontend.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMProfileData.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangDriver.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMOption.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangParse.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMCParser.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangSerialization.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangSema.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangAnalysis.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangEdit.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangRewrite.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangAST.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangLex.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangAPINotes.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMBitReader.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libclangBasic.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMMC.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMCore.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMBinaryFormat.a lib/libswiftOption.a lib/libswiftBasic.a /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMSupport.a -lz -lrt -ldl -ltinfo -lpthread -lm /home/asb/Developer/swift/build/Ninja-DebugAssert/llvm-linux-x86_64/lib/libLLVMDemangle.a lib/libswiftDemangling.a /usr/lib64/libuuid.so && :
lib/libswiftBasic.a(Statistic.cpp.o): In function `swift::FrontendStatsTracer::FrontendStatsTracer(swift::UnifiedStatsReporter*, llvm::StringRef, swift::Decl const*)':
/home/asb/Developer/swift/swift/lib/Basic/Statistic.cpp:462: undefined reference to `swift::UnifiedStatsReporter::TraceFormatter const* swift::FrontendStatsTracer::getTraceFormatter<swift::Decl const*>()'
lib/libswiftBasic.a(Statistic.cpp.o): In function `swift::FrontendStatsTracer::FrontendStatsTracer(swift::UnifiedStatsReporter*, llvm::StringRef, swift::ProtocolConformance const*)':
/home/asb/Developer/swift/swift/lib/Basic/Statistic.cpp:468: undefined reference to `swift::UnifiedStatsReporter::TraceFormatter const* swift::FrontendStatsTracer::getTraceFormatter<swift::ProtocolConformance const*>()'
lib/libswiftBasic.a(Statistic.cpp.o): In function `swift::FrontendStatsTracer::FrontendStatsTracer(swift::UnifiedStatsReporter*, llvm::StringRef, swift::Expr const*)':
/home/asb/Developer/swift/swift/lib/Basic/Statistic.cpp:472: undefined reference to `swift::UnifiedStatsReporter::TraceFormatter const* swift::FrontendStatsTracer::getTraceFormatter<swift::Expr const*>()'
lib/libswiftBasic.a(Statistic.cpp.o): In function `swift::FrontendStatsTracer::FrontendStatsTracer(swift::UnifiedStatsReporter*, llvm::StringRef, clang::Decl const*)':
/home/asb/Developer/swift/swift/lib/Basic/Statistic.cpp:477: undefined reference to `swift::UnifiedStatsReporter::TraceFormatter const* swift::FrontendStatsTracer::getTraceFormatter<clang::Decl const*>()'
lib/libswiftBasic.a(Statistic.cpp.o): In function `swift::FrontendStatsTracer::FrontendStatsTracer(swift::UnifiedStatsReporter*, llvm::StringRef, swift::SILFunction const*)':
/home/asb/Developer/swift/swift/lib/Basic/Statistic.cpp:482: undefined reference to `swift::UnifiedStatsReporter::TraceFormatter const* swift::FrontendStatsTracer::getTraceFormatter<swift::SILFunction const*>()'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/include/swift/Basic/Statistic.h
+++ b/include/swift/Basic/Statistic.h
@@ -205,16 +205,16 @@ public:
   /// entity type, and produce a tracer that's either active or inert depending
   /// on whether the provided \p Reporter is null (nullptr means "tracing is
   /// disabled").
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName);
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName,
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName);
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName,
                       const Decl *D);
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName,
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName,
                       const ProtocolConformance *P);
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName,
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName,
                       const clang::Decl *D);
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName,
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName,
                       const Expr *E);
-  FrontendStatsTracer(UnifiedStatsReporter *Reporter,  StringRef EventName,
+  FrontendStatsTracer(UnifiedStatsReporter *Reporter, StringRef EventName,
                       const SILFunction *F);
 };
 
@@ -225,20 +225,61 @@ public:
 // for those entity types. If you want to trace those types, it's assumed you're
 // linking with the object files that define the tracer.
 
-template<> const UnifiedStatsReporter::TraceFormatter*
+template <>
+const UnifiedStatsReporter::TraceFormatter *
 FrontendStatsTracer::getTraceFormatter<const Decl *>();
 
-template<> const UnifiedStatsReporter::TraceFormatter*
+template <>
+const UnifiedStatsReporter::TraceFormatter *
 FrontendStatsTracer::getTraceFormatter<const ProtocolConformance *>();
 
-template<> const UnifiedStatsReporter::TraceFormatter*
+template <>
+const UnifiedStatsReporter::TraceFormatter *
 FrontendStatsTracer::getTraceFormatter<const clang::Decl *>();
 
-template<> const UnifiedStatsReporter::TraceFormatter*
+template <>
+const UnifiedStatsReporter::TraceFormatter *
 FrontendStatsTracer::getTraceFormatter<const Expr *>();
 
-template<> const UnifiedStatsReporter::TraceFormatter*
+template <>
+const UnifiedStatsReporter::TraceFormatter *
 FrontendStatsTracer::getTraceFormatter<const SILFunction *>();
 
-}
+// Provide inline definitions for the delegating constructors.  These avoid
+// introducing a circular dependency between libParse and libSIL.  They are
+// marked as `inline` explicitly to prevent ODR violations due to multiple
+// emissions.  We cannot force the inlining by defining them in the declaration
+// due to the explicit template specializations of the `getTraceFormatter`,
+// which is declared in the `FrontendStatsTracer` scope (the nested name
+// specifier scope cannot be used to declare them).
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S)
+    : FrontendStatsTracer(R, S, nullptr, nullptr) {}
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S, const Decl *D)
+    : FrontendStatsTracer(R, S, D, getTraceFormatter<const Decl *>()) {}
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S,
+                                                const ProtocolConformance *P)
+    : FrontendStatsTracer(R, S, P,
+                          getTraceFormatter<const ProtocolConformance *>()) {}
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S,
+                                                const clang::Decl *D)
+    : FrontendStatsTracer(R, S, D, getTraceFormatter<const clang::Decl *>()) {}
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S, const Expr *E)
+    : FrontendStatsTracer(R, S, E, getTraceFormatter<const Expr *>()) {}
+
+inline FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R,
+                                                StringRef S,
+                                                const SILFunction *F)
+    : FrontendStatsTracer(R, S, F, getTraceFormatter<const SILFunction *>()) {}
+
+} // namespace swift
 #endif // SWIFT_BASIC_STATISTIC_H

--- a/lib/Basic/Statistic.cpp
+++ b/lib/Basic/Statistic.cpp
@@ -453,35 +453,6 @@ FrontendStatsTracer::FrontendStatsTracer(
 
 FrontendStatsTracer::FrontendStatsTracer() = default;
 
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S)
-    : FrontendStatsTracer(R, S, nullptr, nullptr)
-{}
-
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S,
-                                         const Decl *D)
-    : FrontendStatsTracer(R, S, D, getTraceFormatter<const Decl *>())
-{}
-
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S,
-                                         const ProtocolConformance *P)
-    : FrontendStatsTracer(R, S, P,
-                          getTraceFormatter<const ProtocolConformance *>()) {}
-
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S,
-                                         const Expr *E)
-    : FrontendStatsTracer(R, S, E, getTraceFormatter<const Expr *>())
-{}
-
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S,
-                                         const clang::Decl *D)
-    : FrontendStatsTracer(R, S, D, getTraceFormatter<const clang::Decl *>())
-{}
-
-FrontendStatsTracer::FrontendStatsTracer(UnifiedStatsReporter *R, StringRef S,
-                                         const SILFunction *F)
-    : FrontendStatsTracer(R, S, F, getTraceFormatter<const SILFunction *>())
-{}
-
 FrontendStatsTracer&
 FrontendStatsTracer::operator=(FrontendStatsTracer&& other)
 {

